gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java

```diff
@@ -277,28 +295,45 @@ private void addMapPin(MapPin mapPin) {
   }
 
   private void addMapPolyline(MapPolygon mapPolygon) {
-    for (ImmutableSet<Point> vertices : mapPolygon.getVertices()) {
-      PolylineOptions options = new PolylineOptions();
-
-      // Read-only
-      options.clickable(false);
-
-      // Add vertices to PolylineOptions
-      stream(vertices).map(GoogleMapsMapAdapter::toLatLng).forEach(options::add);
+    PolylineOptions options = new PolylineOptions();
+    // Read-only
+    options.clickable(false);
+    // Add vertices to PolylineOptions
+    ImmutableList<LatLng> vertices = stream(mapPolygon.getVertices())
+        .map(GoogleMapsMapAdapter::toLatLng)
+        .collect(toImmutableList());
+    stream(vertices).forEach(options::add);
+    // Add to map
+    Polyline polyline = map.addPolyline(options);
+    polyline.setTag(mapPolygon);
+    // Style polyline
+    if (!isPolygonCompleted(mapPolygon.getVertices())) {
+      BitmapDescriptor customCap = bitmapDescriptorFromVector();
+      polyline.setStartCap(new CustomCap(customCap));
+      polyline.setEndCap(new CustomCap(customCap));
+    }
+    polyline.setWidth(getPolylineStrokeWidth());
+    polyline.setColor(parseColor(mapPolygon.getStyle().getColor()));
+    polyline.setJointType(JointType.ROUND);
 
-      // Add to map
-      Polyline polyline = map.addPolyline(options);
-      polyline.setTag(mapPolygon);
+    polylines.add(polyline);
+    polygons.put(mapPolygon, vertices);
+  }
 
-      // Style polyline
-      polyline.setStartCap(new RoundCap());
-      polyline.setEndCap(new RoundCap());
-      polyline.setWidth(getPolylineStrokeWidth());
-      polyline.setColor(parseColor(mapPolygon.getStyle().getColor()));
-      polyline.setJointType(JointType.ROUND);
+  private boolean isPolygonCompleted(List<Point> vertices) {
+    return vertices.size() != stream(vertices).distinct().count();
+  }
 
-      polylines.add(polyline);
-    }
+  private BitmapDescriptor bitmapDescriptorFromVector() {
+    Drawable vectorDrawable = ContextCompat.getDrawable(context, R.drawable.ic_endpoint);
+    vectorDrawable
+        .setBounds(4, 4, vectorDrawable.getIntrinsicWidth(), vectorDrawable.getIntrinsicHeight());
```

Comment:
Why `(4, 4)` specifically? It's not clear looking at the code; please either comment or move to dimens.

Reply:
updated